cargo-cyclonedx/src/generator.rs

```diff
@@ -169,8 +169,56 @@ impl SbomGenerator {
         component
     }
 
+    /// Same as [Self::create_component] but also includes information
+    /// on binaries and libraries comprising it as subcomponents
+    fn create_toplevel_component(&self, package: &Package) -> Component {
```

Comment:
I don't think there is, but I believe we should because the binary name may differ from the package name. For example the [`fd` utility](https://crates.io/crates/fd-find) has `fd-find` package name.